### PR TITLE
Menu Driven Circular Doubly Linked List in c++

### LINKED ISSUE #1565: Circular-Doubly-Linked-List
I would like to create a dynamic menu driven program for Circular Doubly Linked List in C++ for GSSoC'2020. Please assign this task to me.

### PR DESCRIPTION
**Pull request by HardevKhandhar**
Fixes #1565 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Menu-driven circular doubly linked list program 
- 

#### Languages Used:
- C++
- 

#### Files Added:
- Circular_Doubly_Linked_list.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp [hunk 1]
```diff
@@ -1,131 +1,412 @@
...
+    struct node *newNode;
+    struct node *trav;
+    trav = HEAD;
+    int value;
+    cout << "Enter the value to be inserted: ";
+    cin >> value;
+    newNode = createNode(value);
+    if(HEAD == NULL)
+    {
+        newNode->lptr = NULL;
```
Comment (somya-kapoor) [inline Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp]: Please add space around ->

File: Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp [hunk 2]
```diff
@@ -1,131 +1,412 @@
...
+{
+    struct node *trav;
+    struct node *next;
+    struct node *newNode;
+    trav = HEAD;
+    int value;
+    cout << "Enter the value to be inserted: ";
+    cin >> value;
+    newNode = createNode(value);
+    while(trav->data != x && trav -> rptr != NULL)
```
Comment (somya-kapoor) [inline Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp]: Please add space around ->

File: Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp [hunk 3]
```diff
@@ -1,131 +1,412 @@
...
+    {
+        trav = trav->rptr;
```
Comment (somya-kapoor) [inline Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp]: Please add space around ->

File: Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp [hunk 4]
```diff
@@ -1,131 +1,412 @@
...
+}
+
+// Delete node from a particular position in the list
+void deleteAtPosition(int x)
+{
+    struct node *trav;
+    struct node *previous;
+    struct node *next;
+    trav = HEAD;
+    while(trav->data != x && trav -> rptr != NULL)
```
Comment (somya-kapoor) [inline Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp]: Please add space around ->

File: Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp [hunk 5]
```diff
@@ -1,131 +1,412 @@
...
+    {
+        previous = trav;
+        trav = trav->rptr;
```
Comment (somya-kapoor) [inline Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.cpp]: Please add space around ->

**Comment by somya-kapoor** [comment]
After making changes please squash the commits

**Comment by HardevKhandhar** [comment]
@somya-kapoor Ma'am

The Changes are updated & commits are squashed.
Thank you.

**Pull request merged**